src/conn/conn_log.c

```diff
@@ -104,19 +104,20 @@ __logmgr_version(WT_SESSION_IMPL *session, bool reconfig)
      * does not mean the direction of change from the release we may be running currently.
      */
     if (conn->compat_major < WT_LOG_V2_MAJOR) {
-        new_version = 1;
         first_record = WT_LOG_END_HEADER;
+        new_version = 1;
         downgrade = true;
     } else {
-        /*
-         * Assume current version unless the minor compatibility setting is the earlier version.
-         */
         first_record = WT_LOG_END_HEADER + log->allocsize;
-        new_version = WT_LOG_VERSION;
-        downgrade = false;
         if (conn->compat_minor == WT_LOG_V2_MINOR) {
             new_version = 2;
             downgrade = true;
+        } else if (conn->compat_minor < WT_LOG_V4_MINOR) {
+            new_version = 3;
+            downgrade = true;
+        } else {
```

Comment:
I'm just wondering the motivation for the organizational change? Why not assume the current version still? (Or really we still are assuming it, but in the 'else' clause instead of the mainline code.)

Reply:
My intent was to be consistent with the logic flow below. Changed back to original organisation.